contracts/staking/ClusterRewards.sol

```diff
@@ -493,15 +477,14 @@ contract ClusterRewards is
 
     bytes32 public constant RECEIVER_PAYMENTS_MANAGER = keccak256("RECEIVER_PAYMENTS_MANAGER");
 
-    mapping(address => uint256) public receiverBalance;
-    mapping(address => uint256) public receiverRewardPerEpoch;
+    mapping(address => ReceiverPayment) public receiverRewardPayment;
 
-    function _increaseReceiverBalance(address staker, uint256 amount) external override onlyRole(RECEIVER_PAYMENTS_MANAGER) {
-        receiverBalance[staker] += amount;
+    function _increaseReceiverBalance(address staker, uint128 amount) external override onlyRole(RECEIVER_PAYMENTS_MANAGER) {
+        receiverRewardPayment[staker].rewardRemaining += amount;
```

Comment:
Inputs can be uint256 instead, but can add checks so that inputs doesn't go above uint128

Reply:
can use uint256, but will have to downcast it.
that will be un-necessary code